### ISSUE TITLE
Startup class (FunctionsStartup) is never called

### ISSUE THREAD

**Issue by Azazel666** [Azure/azure-functions-host#6837]
I'm trying to create a azure function and I'm currently trying to get DI to work but with no success. I have checked [documentations ](https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection) and multiple articles access the web and I don't understand where I go wrong. Essentially I'm trying to setup a startup class derived from FunctionsStartup but the class is never called upon.

#### Repro steps

Provide the steps required to reproduce the problem:

Our startup class looks like this
```csharp
[assembly: FunctionsStartup(typeof(IMD.Cloud.Functions.Startup))]
namespace IMD.Cloud.Functions
{
    public class Startup : FunctionsStartup
    {
        
        public override void Configure(IFunctionsHostBuilder builder)
        {
            
        }
    }
}
````
host.json
````json
{
    "version": "2.0"
}
````
local.settings.json
```json
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "FUNCTIONS_WORKER_RUNTIME": "dotnet",
    "FUNCTIONS_EXTENSION_VERSION": "~3",
  }
}
```

Our cs.proj file has this properties:
```xml
<PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <AzureFunctionsVersion>v3</AzureFunctionsVersion>
    <UserSecretsId>***********************</UserSecretsId>
  </PropertyGroup>
```

But still I can't get breakpoints to trigger in the Startup:Configure(IFunctionsHostBuilder builder) method and any DI setup made here is not working when trying to use DI in classes.

We are using .net core 3.1, Azure functions V3.
Microsoft.NET.Sdk.Functions is running 3.0.9, but I have tested also 3.0.3 & 3.0.5 after reading other issues where it was recommended.
Microsoft.Azure.Functions.Extensions is running 1.1.0, but also tried running 1.0.0.

#### Expected behavior

The method Startup:Configure(IFunctionsHostBuilder builder)  to be called upon for our DI setup.

#### Actual behavior

Startup:Configure(IFunctionsHostBuilder builder)  is never called upon.

#### Known workarounds

Have not made any progress finding a workaround.

**Comment by Azazel666** [comment]
Root cause seems to have been the <UserSecretsId> element of the project file. It casuses it to break and you need to add another nuget package Microsoft.Extensions.Configuration.UserSecrets for it to resume working. I have not seen this being mentioned in any documentation or any of the articles about getting FuctionsStartup to work. But i found another issue with this link to [damienbod](https://damienbod.com/2020/07/12/azure-functions-configuration-and-secrets-management/) and he mentions it there.

**Comment by anthonychu** [comment]
FYI @gzuber as this relates to user secrets.

@Azazel666 Can you share more about the problem you ran into and how you solved it?

**Comment by Azazel666** [comment]
> FYI @gzuber as this relates to user secrets.
> 
> @Azazel666 Can you share more about the problem you ran into and how you solved it?

We are developing a Azure function to be scheduled. As we want to take advantage of DI i wanted to setup the dependencies using the startup class, however we dont want to hardcode in connections strings etc. that will be in version control cause we want to be flexible between developers and not accidently push and overwrite someone else's settings so we use secrets.json for those values that will later be set for the environment in Azure.

I started with initializing secrets for my project cause I knew I was planing on using it.
Then I followed the [documentation](https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection) and implemented my startup class, which of course did not work and i started setting breakpoints which never triggered so I started google and finding all sort of examples but they where all pretty much the same.

I did at one point see some mentioning of `<UserSecretsId>***********************</UserSecretsId>` in your csproj file could be a potential issue but I disregarded it at the time. Then after posting this issue here I continued searching for answers and again found the thing about "UserSecretsId" in csproj file, so I tested and my breakpoints started triggering, so with more help of google I found another issue here with the link to [damienbod](https://damienbod.com/2020/07/12/azure-functions-configuration-and-secrets-management/) article where I finally found the holy grail where he pointed out that a nuget package was needed to get user secrets to work in a azure functions project. However I have not seen any mentioning of these stuff in the official documentations.
